app/models/spree/gateway/amazon.rb

```diff
@@ -183,6 +183,21 @@ def void(response_code, gateway_options)
       return ActiveMerchant::Billing::Response.new(true, "Success", response)
     end
 
+    def cancel(response_code)
+      payment = Spree::Payment.find_by!(response_code: response_code)
+      order = payment.order
+      load_amazon_mws(payment.source.order_reference)
```

Comment:
It feels like once we get the `order`, both `void` and `cancel` could use the exact same code, couldn't they?